lib/sink/connection/client_connection.ex

```diff
@@ -377,19 +381,15 @@ defmodule Sink.Connection.ClientConnection do
     end
   end
 
-  def handle_info({:tcp_closed, _}, state) do
-    {:stop, :normal, state}
-  end
-
-  def handle_info({:ssl_closed, _}, state) do
-    {:stop, :normal, state}
-  end
-
-  def handle_info({:ssl_error, reason}, state) do
-    {:stop, {:error, reason}, state}
+  def handle_info({closed, _}, state) when closed in [:tcp_closed, :ssl_closed] do
+    if State.disconnecting?(state) do
```

Comment:
I think you want to think about this the other way: if the connection was closed when the connection state was `:requesting_connection` then we have a pretty good guess (though this is still a bit of a hack) that the server rejected the connection.

Reply:
I'm not sure that's a reasonable assumption to make. It could just as much be caused by a network issue while sink is in that state. Getting a negative connection response should be the reason for sink to consider itself being rejected. Any negative connection response will result in `State.disconnecting?(state)` being true.

I don't think penalizing a nova on bad network with a 5 minutes timeout would be a good idea and we should rather expect that connection responses are actually received if the nova is not already having a hard time keeping the connection open.